src/codesyncd/handlers/tab_handler.ts

```diff
@@ -0,0 +1,85 @@
+import fs from "fs";
+import os from "os";
+import path from "path";
+
+import { ITabYML, ITabFile, ITabToSend } from "../../interface";
+import {generateSettings} from "../../settings";
+import {readYML} from "../../utils/common";
+import {CodeSyncLogger} from "../../logger";
+import {pathUtils} from "../../utils/path_utils";
+import {initUtils} from "../../init/utils";
+import {VSCODE} from "../../constants";
+import { removeTabFile } from "../../utils/tab_utils";
+
+export class TabHandler {
+    accessToken: string;
+	tabData: ITabYML;
+	tabFilePath: string;
+	repoPath: string;
+
+	repo_id: number;
+	created_at: string;
+	source: string;
+	file_name: string;
+	tabs: ITabFile[];
+
+	configJSON: any;
+    configRepo: any;
+
+	constructor(
+		tabData: ITabYML,
+		tabFilePath: string | null = null,
+		repoPath: string,
+		accessToken: string
+	) {
+		this.accessToken = accessToken;
+		this.tabData = tabData;
+		this.repo_id = tabData.repo_id;
+		this.created_at = tabData.created_at;
+		this.source = tabData.source;
+		this.file_name = tabData.file_name;
+		this.tabs = tabData.tabs;
+		this.repoPath = repoPath
+		if (!tabFilePath) return;
+		this.tabFilePath = tabFilePath;
+
+
+		const settings = generateSettings();
+        this.configJSON = readYML(settings.CONFIG_PATH);
+        this.configRepo = this.configJSON.repos[this.repoPath];
+	}
+
+	createTabToSend() {
+		return {
+			repo_id: this.repo_id,
```

Comment:
We can do this.tabsData.repo_id,
probably no need to define every variable in constructor